js/fileinput.js

```diff
@@ -570,6 +570,7 @@
             if (!self.isPreviewable) {
                 self.showPreview = false;
             }
+            self.uploadUrl = !self.showUpload && !self.uploadUrl ? 'foo' : self.uploadUrl;

```

Comment:
the presence/absence of self.uploadUrl should not determine whether the drag & drop interface be present or not